### PR TITLE
Combine improvements

### PR DESCRIPTION
**Pull request by space-alien**
* All Combine signatures now have an optional error message separator param.
* DRYed up and improved the code
* Added some docs

### CODE REVIEW

File: CSharpFunctionalExtensions/Result/Extensions/CombineAsyncLeft.cs [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 namespace CSharpFunctionalExtensions
```
Comment (golavr) [inline CSharpFunctionalExtensions/Result/Extensions/CombineAsyncLeft.cs]: you can remove the platform divergence in functions using: 

```
#if NET40
using Task = System.Threading.Tasks.TaskEx;
#else
using Task = System.Threading.Tasks.Task;
#endif
```
image

**Comment by golavr** [comment]
to make combine more fluent in some cases
meaning not having to create collection of results or tasks results to be able using combine

you can add:

image



image

...need to add all variations

**Comment by vkhorikov** [comment]
@space-alien looks great, merge as you see fit.

**Comment by space-alien** [comment]
@golavr I like the look of the Task<Result> extension methods too, I'd like to loop back on this when I have a bit more time.

**Pull request merged**